tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -0,0 +1,121 @@
+        retry --wait 1 -n 100 sh -x -c 'systemctl is-active snapd.service snapd.socket'
+    }
+
+    # Check that snapd has restarted and now has a new PID, then set SNAPD_PID to that new PID.
+    check_snapd_restarted() {
+        #shellcheck disable=SC2016
+        retry --wait 1 -n 100 --env SNAPD_PID="$SNAPD_PID" sh -x -c 'systemctl is-active snapd.service snapd.socket && test ! "$(systemctl show --property MainPID snapd.service | cut -f2 -d=)" = "$SNAPD_PID"'
+        # Set SNAPD_PID so future checks use most recent PID
+        SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
+    }
+
+    check_prompting_setting() {
+        # Check that snap CLI reports prompting flag set correctly
+        snap get system experimental.apparmor-prompting | MATCH "$1"
+        # Check that /v2/snaps/system/conf reports prompting flag set correctly
+        curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf | jq -r '.result.experimental."apparmor-prompting"' | MATCH "$1"
+        # Check that /v2/system-info reports prompting correctly
+        curl -sS --unix-socket /run/snapd.socket http://localhost/v2/system-info | jq -r '.result.features."apparmor-prompting".enabled' | MATCH "$1"
+    }
+
+    echo "Precondition check that snapd is active"
+    retry --wait 1 -n 100 sh -x -c 'systemctl is-active snapd.service snapd.socket'
+    SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
+
+    echo "Ensure prompting is initially disabled"
+    if snap get system experimental.apparmor-prompting | grep 'true' ; then
+        snap set system experimental.apparmor-prompting=false
+        check_snapd_restarted
+        check_prompting_setting "false"
+    fi
+
+    echo "Enable prompting via snap client"
+    snap set system experimental.apparmor-prompting=true
+
+    echo "Check that snapd restarted after prompting set to true via snap client"
+    check_snapd_restarted
+    check_prompting_setting "true"
+
+    echo "Disable prompting via snap client"
+    snap set system experimental.apparmor-prompting=false
+
+    echo "Check that snapd restarted after prompting set to false via snap client"
+    check_snapd_restarted
+    check_prompting_setting "false"
+
+    echo "Enable prompting via API request"
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf -X PUT -d '{"experimental.apparmor-prompting": true}' | jq -r '.status' | MATCH "Accepted"
+
+    echo "Check that snapd restarted after prompting set to true via curl"
+    check_snapd_restarted
+    check_prompting_setting "true"
+
+    echo "Disable prompting via API request"
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf -X PUT -d '{"experimental.apparmor-prompting": false}' | jq -r '.status' | MATCH "Accepted"
+
+    echo "Check that snapd restarted after prompting set to false via curl"
+    check_snapd_restarted
+    check_prompting_setting "false"
```

Comment:
FWIW, this repeats what the snap client is doing, not sure if it's worth repeating the same operation

Reply:
Fair point. Since prompting may be interacted with via the snap client or the API directly, I figured testing both would be useful, but the client should do exactly what curl does. I'd be more in favor of testing `curl` though, since that's what the desktop client will use. But really, I think testing both wouldn't hurt.